### ISSUE TITLE
FAIL: runTest (tst_diskless.DisklessTestCase) with some HDF5 options

### ISSUE THREAD

**Issue by ArchangeGabriel** [Unidata/netcdf4-python#669]
Hi there,

I’m currently doing tests on HDF/netCDF packaging for ArchLinux, and I’m having a failure with python-netcdf4 when trying to build using an HDF5 version with C/CXX/Fortran/MPI/HL enabled.

```
FAIL: runTest (tst_diskless.DisklessTestCase)
testing diskless file capability
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/python-netcdf4/src/netcdf4-python-1.2.8rel/test/tst_diskless.py", line 68, in runTest
    assert(os.path.isfile(self.file)==False)
AssertionError

----------------------------------------------------------------------
Ran 93 tests in 17.978s

FAILED (failures=1)
not running tst_unicode.py ...
not running tst_dap.py ...

netcdf4-python version: 1.2.8
HDF5 lib version:       1.10.1
netcdf lib version:     4.4.1.1
numpy version           1.12.1
```

Do you know whether that’s expected and if so what is the culprit? If not I’ll try to build all combinations of HDF5 options to determine that, but it might take some time.

**Comment by ArchangeGabriel** [comment]
Note that this only fails if netCDF4 is built against this HDF5 combo, if I build it against a default HDF5 (C/HL), but then use the full HDF5 one for building python-netcdf4, it seems to work.

**Comment by jswhit** [comment]
This is not something I've seen before, but then again I have not ever tried running the tests with an MPI-enabled build of HDF5.

**Comment by ArchangeGabriel** [comment]
OK, so build against HDF5(C/CXX/Fortran/HL)→netCDF→python is perfectly fine.

So:
- [x] HDF5 →netCDF → python
- [x] HDF5-MPI → netCDF over HDF5 → python over HDF5-MPI
- [ ] HDF5-MPI → netCDF over HDF5-MPI (but serial) → python over HDF5-MPI

I still need to test netCDF built with parallel mode enabled. And also, all the MPI ones with OpenMPI 1.10.6 (my previous tests used this version), 2.1 (current version on my system) and eventually MPICH…

I’ll keep you updated, but I expect this to take me some time.

**Comment by ArchangeGabriel** [comment]
So… I think there was some confusion on my side on which combination gave what, thus I’ve retested a lots of cases. Here are my reports, note I’m always building netcdf4-python with the HDF5-MPI (building both netCDF and netcdf4-python with non-MPI works OK, and it is not possible AFAIK to build netCDF against HDF5-MPI but netcdf4-python against HDF5). netCDF is version 4.4.1.1.

HDF5 1.10.0_patch1, OpenMPI 2.1:
- [x] HDF5-MPI → netCDF over HDF5
- [x] HDF5-MPI → netCDF over HDF5-MPI (but serial)
- [ ] HDF5-MPI → netCDF over HDF5-MPI (parallel build)

HDF5 1.10.1, OpenMPI 2.1:
- [x] HDF5-MPI → netCDF over HDF5
- [x] HDF5-MPI → netCDF over HDF5-MPI (but serial)
- [ ] HDF5-MPI → netCDF over HDF5-MPI (parallel build)

HDF5 1.10.1, OpenMPI 1.10.6:
- [x] HDF5-MPI → netCDF over HDF5
- [x] HDF5-MPI → netCDF over HDF5-MPI (but serial)
- [x] HDF5-MPI → netCDF over HDF5-MPI (parallel build)

So this seems to be an issue with netCDF parallel mode combined with OpenMPI 2.0/2.1. I’ll try again with the 4.5RC release of netCDF, because some things were fixed in this version (I’ll only use HDF5 1.10.1 this time though, but still both MPI of course).

I still need to try MPICH too, but that will come later.

**Comment by ArchangeGabriel** [comment]
The issue was likely fixed by OpenMPI 3.0. I’m thus closing this issue. (BTW, this is now available in ArchLinux in both OpenMPI and serial flavour)
